pvlib/iotools/tmy.py

```diff
@@ -3,9 +3,21 @@
 import datetime
 import re
 import pandas as pd
+import warnings
+from pvlib._deprecation import pvlibDeprecationWarning
 
+# Dictionary mapping TMY3 names to pvlib names
+VARIABLE_MAP = {
+    'GHI': 'ghi',
+    'DNI': 'dni',
+    'DHI': 'dhi',
+    'Pressure': 'pressure',
+    'Wdir': 'wind_direction',
+    'Wspd': 'wind_speed',
+}
```

Comment:
This looks great :)

From taking a quick look at the ``_recolumn()`` function, I think we can probably also add the following. Note I've added the units such that we can map variables directly from the original names and do not have to use the ``_recolumn()`` function as an intermediary step.

@pvlib/pvlib-core Is there a standard name for precipitation (in TMY3 it's called "Lprecip depth (mm)")?

```suggestion
VARIABLE_MAP = {
    'GHI (W/m^2)': 'ghi',
    'DNI (W/m^2)': 'dni',
    'DHI (W/m^2)': 'dhi',
    'Pressure (mbar)': 'pressure',
    'Wdir (degrees)': 'wind_direction',
    'Wspd (m/s)': 'wind_speed',
    'Dry-bulb (C)': 'temp_air',
    'Dew-point (C)': 'temp_dew',
    'RHum (%)': 'relative_humidity',
    'Alb (unitless)': 'albedo',
}
```